### DISCUSSION TITLE
Is there any way to create a button and have its background as a image?

### DISCUSSION THREAD

**Discussion by deven740** [pymupdf/PyMuPDF#2148] [Looking for help]
Hi everyone!

Can I add a Image as a background for button I created using 

`fitz.PDF_WIDGET_TYPE_BUTTON`

Also this method creates a checkbox instead of a normal button.

 is there any way to set Image as a background for the button?
 
 Thanks in advance.

**Comment by JorjMcKie** [comment]
> is there any way to set Image as a background for the button?

No, there is no way to do that, sorry.

**Reply by deven740** [reply]
Okay, thanks for the reply.

Can you tell me why a checkbox is created instead of a normal button using below code?

`import fitz

doc = fitz.open("blank.pdf") #Open document
p = doc[0] #get the page object on page 1
widget = fitz.Widget() #create a widget
widget.rect = fitz.Rect(10,10,50,50) #set the size and position (x1, y1, x2, y2)
widget.field_type = fitz.PDF_WIDGET_TYPE_BUTTON
widget.field_flags = fitz.PDF_BTN_FIELD_IS_PUSHBUTTON #set the widget become push button
widget.field_name = "btn"
annot = p.add_widget(widget) # add the widget to the page
doc.save("t2.pdf") #output the file`

**Reply by JorjMcKie** [reply]
This is a bug! Based on your initial comment, I rechecked and can confirm the wrong behavior.
Fix was easy and will be included in the next version.
If you use Windows or Linux Python 3.10, I can give a pre-version wheel ...
